### PR TITLE
support test failed as error and test naming ending with #id* in tap file

### LINKED ISSUE #407: Support failed test case name ending with #id* as part of tap test results
In most cases the openjdk test name is ended with .java or .sh. There are cases the name ending with #id*. We need to support this case as test results summary in tap file.

This is noticed by https://github.com/eclipse-openj9/openj9/issues/16763

### LINKED ISSUE #408: Support openjdk error test results as part of tap results
Some openjdk test status is error instead of failed, which is also need to be triaged. This error case should be supported in tap results file and hence tests can be rerun by rerun jdk_custom or hotspot_custom link.

### PR DESCRIPTION
**Pull request by sophia-guo**
Support test result as error
Test results: passed: 105; failed: 7; error: 1

test naming ending with #id*
TEST: java/lang/Thread/virtual/stress/PinALot.java#id0

Fix #407 
Fix #408 

Signed-off-by: Sophia Guo <sophia.gwf@gmail.com>

### CODE REVIEW

**Comment by sophia-guo** [comment]
Grinder https://ci.adoptium.net/view/Test_grinder/job/Grinder/6736/

**Comment by sophia-guo** [comment]
Grinder reproduce the scenario and failed testcase is printed in tap file and rerun link enabled.https://ci.adoptium.net/job/Grinder/6740/

@llxia

File: .project [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+<projectDescription>
+	<name>TKG</name>
+	<comment></comment>
+	<projects>
+	</projects>
+	<buildSpec>
+	</buildSpec>
+	<natures>
+	</natures>
+</projectDescription>
```
Comment (llxia) [inline .project:11]: Could you remove .project file?

Comment (sophia-guo) [reply .project]: Updated

**Comment by karianna** [comment]
@sophia-guo I assume this is the usual email address registerd with the EF CLA ?

**Review by karianna — APPROVED** [review]

**Review by smlambert — APPROVED** [review]

**Pull request merged**